src/odb.rs

```diff
@@ -218,6 +218,20 @@ impl<'repo> Odb<'repo> {
             Ok(())
         }
     }
+
+    /// Add a mempack backend with the given priority. Higher values give the
+    /// backend higher precedence. The default loose and pack backends have
+    /// priorities 1 and 2 respectively (hard-coded in libgit2).
+    pub fn add_mempack_backend(&self, mempack: &Mempack, priority: i32) -> Result<(), Error> {
+        unsafe {
+            try_call!(raw::git_odb_add_backend(
+                self.raw,
+                mempack.raw(),
+                priority as c_int
+            ));
```

Comment:
Like below, can you check to see if it's safe to drop `Mempack` after this call?

Reply:
This one is a bit weird. It looks like dropping the ref on the rust side is safe, because that won't actually free the C struct. In fact, the only thing that frees the C struct is the odb itself, [here](https://github.com/libgit2/libgit2/blob/a83fd5107879d18b31ff8173ea062136256321be/src/odb.c#L690). So what could happen is if the rust code creates a mempack, **doesn't** attach it to an odb, and then drops it, it will get leaked. In that sense maybe it would be better to only allow the mempack to be created from an `Odb` object, and have it automatically added to that `Odb` so that even if the rust side drops the ref, it won't get leaked.

Reply:
I guess the other problem is that as-is, a rust user could create a mempack, add it to an odb, and then keep a ref to the mempack even after the odb is destroyed. That would be bad, as it would be a dangling pointer to freed memory.

I'll rework this so that the mempack can only be created from an odb and the lifetime is more tightly scoped.